api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +561,120 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      this.crs = crs;
+    }
+
+    public static GeometryType get() {
+      return of("");
+    }
+
+    public static GeometryType of(String crs) {
+      return new GeometryType(crs == null ? "" : crs);
+    }
+
+    @Override
+    public TypeID typeId() {
+      return TypeID.GEOMETRY;
+    }
+
+    public String crs() {
```

Comment:
maybe can annotate NotNull?

Reply:
The `NotNull` annotation requires an additional dependency, and I noticed that it is not currently used in iceberg-api or iceberg-core. Would you like me to add the additional dependency and annotate this method?

Reply:
Got it, nvm it may not be worth it

Reply:
Iceberg does not generally use nullability annotations.